pkg/cliplugins/workspace/cmd/workspace.go

```diff
@@ -0,0 +1,570 @@
+	workspaceDirectoryConfigOverrideFlags.ClusterOverrideFlags.CertificateAuthority.Description += descriptionSuffix
+	workspaceDirectoryConfigOverrideFlags.ClusterOverrideFlags.InsecureSkipTLSVerify.Description += descriptionSuffix
+	workspaceDirectoryConfigOverrideFlags.ClusterOverrideFlags.TLSServerName.Description += descriptionSuffix
+
+	workspaceDirectoryConfigOverrideFlags.CurrentContext.Description += descriptionSuffix
+	workspaceDirectoryConfigOverrideFlags.CurrentContext.Default = "workspace-directory"
+	workspaceDirectoryConfigOverrideFlags.Timeout.LongName = ""
+
+	clientcmd.BindOverrideFlags(o.workspaceDirectoryOverrides, cmd.Flags(), workspaceDirectoryConfigOverrideFlags)
+	o.configAccess = clientcmd.NewDefaultClientConfigLoadingRules()
+
+	var err error
+	o.startingConfig, err = o.configAccess.GetStartingConfig()
+	if err != nil {
+		return err
+	}
+	return nil
+}
+
+// Validate ensures that all required arguments and flag values are provided
+func (o *WorkspaceOptions) Validate() error {
+	return nil
+}
+
+func (o *WorkspaceOptions) ensureWorkspaceDirectoryContextExists() error {
+	currentContext := o.startingConfig.Contexts[o.startingConfig.CurrentContext]
+
+	workspaceDirectoryContext := o.startingConfig.Contexts[o.workspaceDirectoryOverrides.CurrentContext]
+	workspaceDirectoryCluster := o.startingConfig.Clusters[o.workspaceDirectoryOverrides.CurrentContext]
+	if workspaceDirectoryContext == nil || workspaceDirectoryCluster == nil {
+		if currentContext != nil {
+			workspaceDirectoryCluster = o.startingConfig.Clusters[currentContext.Cluster].DeepCopy()
+			workspaceDirectoryContext = &api.Context{
+				Cluster: o.workspaceDirectoryOverrides.CurrentContext,
+			}
+		} else {
+			workspaceDirectoryCluster = &api.Cluster{}
+			workspaceDirectoryContext = &api.Context{
+				Cluster: o.workspaceDirectoryOverrides.CurrentContext,
+			}
+		}
+		workspaceDirectoryCluster.Server = o.workspaceDirectoryOverrides.ClusterInfo.Server
+		if o.workspaceDirectoryOverrides.ClusterInfo.CertificateAuthority != "" {
+			workspaceDirectoryCluster.CertificateAuthority = o.workspaceDirectoryOverrides.ClusterInfo.CertificateAuthority
+		}
+		if o.workspaceDirectoryOverrides.ClusterInfo.TLSServerName != "" {
+			workspaceDirectoryCluster.TLSServerName = o.workspaceDirectoryOverrides.ClusterInfo.TLSServerName
+		}
+		if o.workspaceDirectoryOverrides.ClusterInfo.InsecureSkipTLSVerify {
+			workspaceDirectoryCluster.InsecureSkipTLSVerify = o.workspaceDirectoryOverrides.ClusterInfo.InsecureSkipTLSVerify
+		}
+		o.startingConfig.Clusters[o.workspaceDirectoryOverrides.CurrentContext] = workspaceDirectoryCluster
+		o.startingConfig.Contexts[o.workspaceDirectoryOverrides.CurrentContext] = workspaceDirectoryContext
+
+		return clientcmd.ModifyConfig(o.configAccess, *o.startingConfig, true)
+	}
+	return nil
+}
+
+func PrioritizedAuthInfo(values ...*api.AuthInfo) *api.AuthInfo {
```

Comment:
needs go doc.

Why is this a public func? Used elsewhere?

Reply:
Nowhere you're right. I'll change that.